symbolic-demangle/src/lib.rs

```diff
@@ -48,38 +50,63 @@ extern "C" {
     fn symbolic_demangle_is_swift_symbol(sym: *const c_char) -> c_int;
 }
 
-/// Defines the output format for demangling.
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum DemangleFormat {
-    /// Strips parameter names (in Swift) and sometimes packages or namespaces.
-    Short,
-
-    /// Outputs the full demangled name including arguments, return types, generics and packages.
-    Full,
-}
-
 /// Options for [`Demangle::demangle`].
 ///
+/// The resulting output depends very much on the language of the mangled [`Name`], and not all
+/// options are fully supported by each language.
+///
+/// The default will output a very simple demangling, which means only the function name in most
+/// cases.
+/// Turning on all the options will result in a fully demangled type.
```

Comment:
What's the motivation for leaving everything off by default?  Perhaps that could also be part of the documentation.  And why is there no simple way of enabling everything?